cmd/snap-update-ns/change.go

```diff
@@ -138,26 +138,48 @@ func NeededChanges(currentProfile, desiredProfile *mount.Profile) []*Change {
 		desiredMap[desired[i].Dir] = &desired[i]
 	}
 
+	// Indexed by mount point path.
+	var reuse map[string]bool = make(map[string]bool)
+	// Indexed by entry ID
+	var desiredIDs map[string]bool = make(map[string]bool)
+	var skipDir string
+
+	// Collect the IDs of desired changes.
+	// We need that below to keep implicit changes from the current profile.
+	for i := range desired {
+		desiredIDs[XSnapdEntryID(&desired[i])] = true
```

Comment:
It's slightly baffling that XSnapdEntryID and XSnapdParentID can return empty strings if missing, but we don't seem to handle that here. Is this guaranteed not to happen? What happens if we have an antry with snapd.parent-id=e1, but "e1" doesn't exist? Would it make sense to have a test for this?

Reply:
I changed the approach to make mount IDs implicit. This is actually both easier to handle and works better in practice where existing desired mount profiles don't have them.